siquery_tables/src/windows/users.rs

```diff
@@ -0,0 +1,332 @@
+            NetUserEnum(ptr::null(),
+                        dw_user_info_level,
+                        0 as DWORD,
+                        user_buffer.as_mut_ptr() as *mut _ as *mut _,
+                        MAX_PREFERRED_LENGTH,
+                        dw_num_users_read,
+                        dw_total_users,
+                        resume_handle)
+        };
+
+        if (ret == NERR_SUCCESS || ret == ERROR_MORE_DATA) &&
+            user_buffer.as_mut_ptr() != ptr::null_mut() {
+
+            let mut iter_buff: LPUSER_INFO_3 = unsafe { ptr::read(user_buffer.as_mut_ptr() as *mut _) };
+
+            for _i in 0..unsafe { *dw_num_users_read }  {
+                let mut user = Users::new();
+                let mut dw_detailed_user_info_level: c_ulong = 4;
+                let mut user_lvl_4buff: Vec<*mut u8> = Vec::with_capacity((mem::size_of::<USER_INFO_4>()) as usize);
+
+                ret = unsafe {
+                    NetUserGetInfo(ptr::null(),
+                                   (*iter_buff).usri3_name,
+                                   dw_detailed_user_info_level,
+                                   user_lvl_4buff.as_mut_ptr())
+                };
+
+                if ret != NERR_SUCCESS || user_lvl_4buff.as_mut_ptr() == ptr::null_mut() {
+                    unsafe {
+                        if user_lvl_4buff.as_mut_ptr() != ptr::null_mut() {
+                            NetApiBufferFree(*user_lvl_4buff.as_mut_ptr() as *mut c_void);
+                        }
+                        println!("with error code {:?}", ret);
+
+                        iter_buff = iter_buff.add(1) as *mut _;
+                    }
+                    continue;
+                }
+                // Will return empty string on fail
+                let mut lp_user_info_4: LPUSER_INFO_4 = unsafe { ptr::read(user_lvl_4buff.as_mut_ptr() as _) };
+                let mut sid: *mut c_void = unsafe { (*lp_user_info_4).usri4_user_sid };
+                unsafe {
+                    if let Ok(username) = lpwstr_to_string((*iter_buff).usri3_name) {
+                        user.username = username;
+                    }
+                    if let Ok(description) = get_user_description((*lp_user_info_4).usri4_comment) {
+                        user.description = description;
+                    }
+                }
+                user.shell = "C:\\Windows\\System32\\cmd.exe".to_string();
+                user.type_ = "local".to_string();
+                if let Ok(sid_string) = sid_to_string(sid) {
+                    user.uuid = sid_string.clone();
+                    processed_sid.push(sid_string.clone());
+                    user.directory = get_user_home_dir(sid_string);
+                }
+                unsafe {
+                    user.uid = (*iter_buff).usri3_user_id as i64;
+                    user.gid = (*iter_buff).usri3_primary_group_id as i64;
+                    user.uid_signed = user.uid;

```

Comment:
Use "as i32 as i64" instead to obtain the signed gid and uid values, then output to i64.